### PR TITLE
Airbyte-ci: Allow airbyte-ci to run from anywhere in project

### LINKED ISSUE #31250: Update `airbyte-ci` to allow running from anywhere in `airbyte` repo

### PR DESCRIPTION
**Pull request by bnchrch**
## Problem
You must run airbyte-ci from root 

closes #31250

## Solution
Just change the working directory when you run airbyte-ci to be root

### Why changing the working directory instead of refactoring for absolute paths
Using absolute paths seemed more prone to error and a larger refactor

Example here: https://github.com/airbytehq/airbyte/pull/31409/files

### Why do you set the working directory in two places
Because `dagger run` requires it set before invoked

and `airbyte-ci-internal` when run on its own also needs it set

### CODE REVIEW

**Comment by erohmensing** [comment]
Hm, I'm afraid that being booted out of your working directory might not really solve the problem here - it still slows down the interaction cycle pretty significantly. That's an assumption, but something that might be worth asking our most receptive connector devs about. 

If we could get around the setting it teice, is there a way we could change directory back to where we were at the end? Maybe some sort of `with` that keeps the context of where we were to get us back there?

File: airbyte-ci/connectors/pipelines/pipelines/commands/airbyte_ci.py [hunk 1]
```diff
@@ -64,6 +66,27 @@ def get_latest_version() -> str:
     raise Exception("Could not find version in pyproject.toml. Please ensure you are running from the root of the airbyte repo.")
 
 
+def get_airbyte_repo() -> git.Repo:
+    """Get the airbyte repo."""
+    return git.Repo(search_parent_directories=True)
```
Comment (evantahler) [inline airbyte-ci/connectors/pipelines/pipelines/commands/airbyte_ci.py:71]: ~~Can we throw if we aren't in a git repo at all?~~
That's done by the caller below 👍

Comment (evantahler) [reply airbyte-ci/connectors/pipelines/pipelines/commands/airbyte_ci.py:71]: Should we validate that the git repo we are in is, in fact, `airbytehq/airbyte`?

File: airbyte-ci/connectors/pipelines/pipelines/commands/airbyte_ci.py [hunk 2]
```diff
@@ -180,6 +203,7 @@ def airbyte_ci(
 airbyte_ci.add_command(connectors)
 airbyte_ci.add_command(metadata)
 airbyte_ci.add_command(test)
+set_working_directory_to_root()
```
Comment (evantahler) [inline airbyte-ci/connectors/pipelines/pipelines/commands/airbyte_ci.py:206]: Probably fine, but do any child-commands or forks need to be in a certain directory?

File: airbyte-ci/connectors/pipelines/pipelines/commands/airbyte_ci.py [hunk 3]
```diff
@@ -64,6 +66,27 @@ def get_latest_version() -> str:
...
+    except git.exc.InvalidGitRepositoryError:
+        logging.warning("Could not find the airbyte repo, falling back to the current working directory.")
+        path = Path.cwd()
+        logging.warning(f"Using {path} as the airbyte repo path.")
+        return path
+
+
+def set_working_directory_to_root() -> None:
+    """Set the working directory to the root of the airbyte repo."""
+    os.chdir(get_airbyte_repo_path_with_fallback())
```
Comment (evantahler) [inline airbyte-ci/connectors/pipelines/pipelines/commands/airbyte_ci.py:87]: nit: if we are changing the CWD, we should probably log something, like "Using /path/to/root as working directory"

**Review by evantahler — APPROVED** [review]
Excellent DX improvement.  My comments are all nits.

File: airbyte-ci/connectors/pipelines/pipelines/commands/airbyte_ci.py (hunk 2 above)
Comment (bnchrch) [reply airbyte-ci/connectors/pipelines/pipelines/commands/airbyte_ci.py:206]: Nope. Currently every `airbyte-ci` command is imported here and every airbyte-ci logic assumes `airbyte/` as the working directory.

**Comment by bnchrch** [comment]
@erohmensing I think theres a small misunderstanding about what `os.chdir` does.

Essentially each process has its own "working directory" meaning that the python program we run, wont change the directory that the terminal is targeting.

Its both why we dont have to reset the working directory back to the orginal and why I have to call `set_working_directory_to_root()` in two places, as


1. The terminal has its own working directory
2. and `Dagger run` (e.g. `airbyte-ci`) has its own working directory
3. and `airbyte-ci-internal` does too.

image

**Comment by erohmensing** [comment]
@bnchrch thanks for the explanation, that makes sense! I'll admit I was only going off of the PR description, I hadnt checked it out and tried it.

File: airbyte-ci/connectors/pipelines/pipelines/commands/airbyte_ci.py [hunk 4]
```diff
@@ -63,6 +67,55 @@ def get_latest_version() -> str:
...
+            return True
+
+    warning_message = f"""
+    ⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️
+
+    It looks like you are not running this command from the airbyte repo ({expected_repo_name}).
+
+    If this command is run from outside the airbyte repo, it will not work properly.
+
+    Please run this command your local airbyte project.
```
Comment (erohmensing) [inline airbyte-ci/connectors/pipelines/pipelines/commands/airbyte_ci.py:84]:
```suggestion
    Please run this command in your local airbyte project.
```

File: airbyte-ci/connectors/pipelines/pipelines/commands/airbyte_ci.py [hunk 5]
```diff
@@ -63,6 +67,55 @@ def get_latest_version() -> str:
...
+
+    ⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️
+    """
+
+    logging.warning(warning_message)
+
+    return False
```
Comment (erohmensing) [inline airbyte-ci/connectors/pipelines/pipelines/commands/airbyte_ci.py:91]: What does this do? Looks like the return value is not used

**Review by erohmensing — APPROVED** [review]
Nice!

**Pull request merged**